dll/win32/browseui/explorerband.cpp

```diff
@@ -427,11 +472,17 @@ LRESULT CExplorerBand::OnContextMenu(UINT uMsg, WPARAM wParam, LPARAM lParam, BO
         goto Cleanup;
     }
 
-    x = LOWORD(lParam);
-    y = HIWORD(lParam);
-    if (x == -1 && y == -1)
+    pt.x = LOWORD(lParam);
+    pt.y = HIWORD(lParam);
+    if ((UINT)lParam == (UINT)-1)
     {
-        // TODO: grab position of tree item and position it correctly
+        RECT r;
+        if (TreeView_GetItemRect(m_hWnd, item, &r, TRUE))
+        {
+            pt.x = r.left + ((r.right - r.left) / 2);
+            pt.y = r.top + ((r.bottom - r.top) / 2);
```

Comment:
I'm placing the menu point in the center of the **item rectangle** to avoid having to think about RTL.

Reply:
Yes I understood you placed the point in the middle; I was asking about the chosen formulae.

Reply:
Mostly because it's easy to interpret what the desired value of y is. I don't know if they compile to the same thing or if yours is better. At least with yours I feel it needs an extra comment about the intention.